script/spider.py

```diff
@@ -41,52 +82,55 @@ def main():
     now = datetime.datetime.now()
     log_info('[*] {date} - Fetching today\'s books'.format(date=now.strftime("%Y-%m-%d %H:%M")))
 
+    packtpub = None
+
     try:
-        #ip_address()
         config = config_file(args.config)
-        types = parse_types(args)
 
-        packpub = Packpub(config, args.dev)
-        packpub.run()
-
-        if args.dev:
-            log_json(packpub.info)
-
-        log_success('[+] book successfully claimed')
-
-        upload = None
-
-        if not args.claimOnly:
-            packpub.download_ebooks(types)
-
-            if args.extras:
-                packpub.download_extras()
-
-            if args.archive:
-                raise NotImplementedError('not implemented yet!')
-
-            if args.upload is not None:
-                upload = Upload(config, args.upload)
-                upload.run(packpub.info['paths'])
-
-            if upload is not None and upload is not SERVICE_DRIVE:
-                log_warn('[-] skip store info: missing upload info')
-            elif args.store is not None:
-                Database(config, args.store, packpub.info, upload.info).store()
-
-        if args.notify:
-            upload_info = None
-
-            if upload is not None:
-                upload_info = upload.info
-
-            Notify(config, packpub.info, upload_info, args.notify).run()
+        #ip_address()
+        log_info('[*] getting daily free ebook')
+
+        try:
+            packpub = Packpub(config, args.dev)
+            run(packpub, args, config)
+        except Exception as e:
+            log_debug(e)
+            if args.notify:
+                Notify(config, None, None, args.notify).sendError(e, 'daily')
+
+        lastNewsletterUrlPath = 'config/lastNewsletterUrl.csv'
+        lastNewsletterUrl = None
+
+        if os.path.isfile(lastNewsletterUrlPath):
+            with open(lastNewsletterUrlPath, 'r') as f:
+                lastNewsletterUrl = f.read()
+
+        currentNewsletterUrl = requests.get('https://goo.gl/kUciut').text
+
+        if lastNewsletterUrl != currentNewsletterUrl:
+            log_info('[*] getting free ebook from newsletter')
+            try:
+                packpubNewsletter = PacktpubFromNewsletter(config, "/packt/free-ebook/javascript-high-performance", args.dev)
```

Comment:
The url is different for each book, so it has to be parsed. But it could be configurable. I will add everything needed to parse the url from the mail to the readme, so everyone can set it up themselves if wanted.